examples/memcached.md

```diff
@@ -0,0 +1,197 @@
+# memcached
+
+As shown in [this blog
+post](https://memcached.org/blog/persistent-memory/), memcached can
+efficiently utilize PMEM as replacement for DRAM. This makes it
+possible to increase cache sizes and/or reduce costs. The memcached
+maintainers make container images available which support this
+feature. This example shows how to take advantage of PMEM with
+memcached running on Kubernetes.
+
+## PMEM as DRAM replacement
+
+Memcached can map PMEM into its address space and then store the bulk
+of the cache there. The advantage is higher capacity and (depending on
+pricing) less costs.
+
+The cache could be stored in a persistent volume and reused after a
+restart (see next section), but when that isn't the goal, a
+[deployment with one CSI ephemeral inline
+volume](../deploy/kustomize/memcached/ephemeral/memcached-ephemeral.yaml)
+per memcached pod is very simple:
+- A Kubernetes deployment manages the memcached instance(s).
+- In the pod template, one additional PMEM volume of a certain size is
+  requested.
+- An init container prepares that volume for use by memcached:
+  - It determines how large the actual file can be that memcached will use.
+  - It changes volume ownership so that memcached can run as non-root
+    process.
+- A wrapper script adds the `--memory-file` and `--memory-limit` parameters
+  when invoking memcached, which enables the use of PMEM.
+
+For this to work, the PMEM-CSI driver must have been deployed with
+`pmem-csi.intel.com` as driver name. No storage classes are
+needed. That name and other parameters in the deployment can be
+modified with
+[`kustomize`](https://github.com/kubernetes-sigs/kustomize). Here's
+how one can change the namespace, volume size or add additional
+command line parameters:
+
+```
+$ mkdir -p my-memcached-deployment
+$ cat >my-memcached-deployment/kustomization.yaml <<EOF
+namespace: demo
+bases:
+ - ../deploy/kustomize/memcached/ephemeral
+patchesStrategicMerge:
+- update-deployment.yaml
+EOF
+$ cat >my-memcached-deployment/update-deployment.yaml <<EOF
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: pmem-memcached
+spec:
+  template:
+    spec:
+      containers:
+      - name: memcached
+        env:
+        - name: MEMCACHED_ARGS
+          value: --conn-limit=500
+      volumes:
+      - name: data-volume
+        csi:
+          volumeAttributes:
+            size: 500Mi
+EOF
+$ kubectl create ns demo
+$ kubectl create --kustomize my-memcached-deployment
+```
+
+Eventually, there will be one pod with memcached running which we can
+connect to with port forwarding through kubectl:
+```
+$ kubectl wait --for=condition=Ready -n demo pods -l app.kubernetes.io/name=pmem-memcached
+pod/pmem-memcached-5f57bcfd58-qjz74 condition met
+$ pod=$(kubectl get -n demo pods -l app.kubernetes.io/name=pmem-memcached -o jsonpath={..metadata.name})
+$ kubectl port-forward -n demo $pod 11211 &
```

Comment:
When I did the above on Clear Linux, I had to hit enter twice to properly see my prompt. Otherwise, after the $ I had  `$ Forwarding from 127.0.0.1:11211 -> 11211` which looked like a command I was about to run.  Hitting enter brought me back to a normal looking prompt.